docs/cpp/source/notes/tensor_cuda_stream.rst

```diff
@@ -0,0 +1,189 @@
+Tensor CUDA Stream API 
+======================
+
+A `CUDA Stream`_ is a linear sequence of execution that belongs to a specific device.
+Pytorch C++ API has defined a CUDAStream class and a few useful functions to support CUDA Stream operations.
```

Comment:
What if this sentence was a little more positive?

"The PyTorch C++ API supports CUDA streams with the CUDAStream class and useful helper functions to make streaming operations easy."